src/ydata_synthetic/synthesizers/gan.py

```diff
@@ -22,6 +22,9 @@
 TrainParameters = namedtuple('TrainParameters', _train_parameters, defaults=('', None, 300, 50, None))
 
 class BaseModel():
+
+    __MODEL__='BaseModel'
```

Comment:
I would either set this property no None or not have it on the BaseModel.

Reply:
Between validating the existence of "__Model__" or setting it everywhere I have chosen the latter. 
But I understand your point, I'll remove it.

Reply:
if we keep it as '' that works same. Without the need for validations I think

Reply:
yes it works